--- a/lib/active_support/core_ext/class/subclasses.rb	2010-08-30 14:14:45.000000000 -0700
+++ b/lib/active_support/core_ext/class/subclasses.rb	2010-08-30 14:13:33.000000000 -0700
@@ -14,6 +14,15 @@
       end
       descendants
     end
+  elsif defined? Maglev
+    def descendents
+      subclasses = []
+      ObjectSpace.loaded_classes do |k|
+        subclasses << k if k < self
+      end
+      subclasses.uniq!
+      subclasses
+    end
   else # MRI
     begin
       ObjectSpace.each_object(Class.new) {}
