--- a/lib/active_support/values/time_zone.rb	2010-08-30 14:37:36.000000000 -0700
+++ b/lib/active_support/values/time_zone.rb	2010-08-30 14:36:37.000000000 -0700
@@ -352,8 +352,14 @@
           when String
           begin
             zones_map[arg] ||= lookup(arg).tap { |tz| tz.utc_offset }
-          rescue TZInfo::InvalidTimezoneIdentifier
-            nil
+          # GemStone: Patch around Trac 757
+          rescue Exception => e
+            if TZInfo::InvalidTimezoneIdentifier === e
+              nil
+            else
+              raise
+            end
+          # End GemStone
           end
           when Numeric, ActiveSupport::Duration
             arg *= 3600 if arg.abs <= 13
