diff --git a/activesupport/lib/active_support/core_ext/class/subclasses.rb b/activesupport/lib/active_support/core_ext/class/subclasses.rb
index bbd8f5a..356f579 100644
--- a/activesupport/lib/active_support/core_ext/class/subclasses.rb
+++ b/activesupport/lib/active_support/core_ext/class/subclasses.rb
@@ -16,6 +16,15 @@ class Class #:nodoc:
       __subclasses__.each {|k| subclasses << k; subclasses.concat k.descendents }
       subclasses
     end
+  elsif defined? Maglev
+    def descendents
+      subclasses = []
+      ObjectSpace.loaded_classes do |k|
+        subclasses << k if k < self
+      end
+      subclasses.uniq!
+      subclasses
+    end
   else
     # MRI
     begin
 
