--- gems/rack-mount-0.6.12/lib/rack/mount/route_set.rb	2010-08-23 10:28:59.000000000 -0700
+++ gems/rack-mount-x/lib/rack/mount/route_set.rb	2010-08-23 10:28:44.000000000 -0700
@@ -328,7 +328,11 @@
             k = keys.map { |key| block.call(key, index) }
             Utils.pop_trailing_blanks!(k)
             k.map! { |key| key || /.*/ }
-            graph[*k] = route
+            # MagLev: Patch around Trac 782
+            #graph[*k] = route
+            k << route
+            graph.store(*k)
+            # End MagLev
           end
         end
         graph
